aten/src/ATen/gen_backend_select_register.py

```diff
@@ -0,0 +1,56 @@
+# This script generates BackendSelectRegister.cpp which is being used for dispatching purposes.
+# We process only those factory functions that have 'backend_select' flag in its native_functions.yaml definition.
+
+from code_template import CodeTemplate
+
+GENERATED_COMMENT = CodeTemplate(
+    "@" + "generated from ${filename}")
+
+FUNCTION_REGISTRATION = CodeTemplate("""\
+.op(torch::RegisterOperators::options()
+  .schema("${schema_string}")
+  .impl_unboxedOnlyKernel<decltype(${function_name}), &${function_name}>(DispatchKey::BackendSelect)
+  .aliasAnalysis(AliasAnalysisKind::FROM_SCHEMA))
+""")
+
+FUNCTION_DEFINITION = CodeTemplate("""\
+Tensor ${function_name}(${method_formals}) {
+  DispatchKey key = options.computeDispatchKey();
+  static auto op = c10::Dispatcher::singleton().findSchemaOrThrow("aten::${name}", "${overload_name}");
+  return op.callUnboxedWithDispatchKey<${formals_types}>(key, ${type_method_actuals});
+}
+""")
+
+def register_backend_select_methods(declarations, template_path, file_manager):
+    backend_select_method_definitions = []
+    backend_select_function_registrations = []
+
+    for decl in declarations:
+        for option in decl["options"]:
+            if '_like' in option['name'] or 'new_' in option['name']:
+                continue
+
+            name = option['name']
+            if 'overload_name' in option and option['overload_name'] != '':
+                name = "{0}_{1}".format(name, option['overload_name'])
+
+            if any("dynamic_type" in a and a["dynamic_type"] == "TensorOptions" for a in option["arguments"]):
```

Comment:
Thanks for the update on why you guys decided to stick with the pattern matching trigger for now, makes sense to keep the scope of the PR manageable. 

My only suggestion would be to clarify what's going on by factoring the triggering condition out into its own function, i.e. something like (note the ... in the comment, where I don't know the exact reasoning but it'd be good to have it here):
```
def needs_backend_select(decl):
    # currently, we register an op under the BackendSelect dispatch key
    # if a TensorOptions argument has been gathered from its declared args
    # (excluding some ops for ... reasons). 

    if '_like' in decl['name'] or 'new_' in decl['name']:
        return false
    return any(a.get('dynamic_type') == 'TensorOptions' for a in decl('arguments'))
```

Reply:
done!